src/basic/path-util.c

```diff
@@ -814,3 +816,57 @@ bool is_device_path(const char *path) {
                 path_startswith(path, "/dev/") ||
                 path_startswith(path, "/sys/");
 }
+
+int systemd_installation_has_version(const char *root, unsigned minimal_version) {
+        _cleanup_free_ char *path = NULL;
+        _cleanup_strv_free_ char **names = NULL;
+        char **name;
+        int r;
+
+        /* Try to guess if systemd installation is later than the specified version. This
+         * is hacky and likely to yield false negatives, particularly if the installation
+         * is non-standard. False positives should be relatively rare.
+         */
+
+        path = prefix_root(root, "lib/systemd/libsystemd-shared-*.so");
```

Comment:
Yeah, i think Gentoo again is doing things differently. Generally, lib64 is understood as the place to put libraries for 64bit if both 32bit and 64bit are to be supported. Except on Gentoo, where lib64 is understood as the place to place 64bit libraries, full stop.

Gentoo's scheme is borked, but if you want to cover that, you'd also have to check for lib64. Now I personally don't care about Gentoo I must say, so the patch is actually fine, but heh...